File: src/lib/compiler/template.js
/**
 * Simple template resolver: replaces {{variable}} with config values.
 * Supports dot notation: {{project.name}}, {{commands.boardCli}}
 */
export function resolveTemplate(template, config) {
  return template.replace(/\{\{([^}]+)\}\}/g, (match, key) => {
    const trimmed = key.trim();
    const value = getNestedValue(config, trimmed);
    if (value === undefined) return match;
    return String(value);
  });
}

function getNestedValue(obj, key) {
  const parts = key.split('.');
  let current = obj;
  for (const part of parts) {
    if (current === null || current === undefined) return undefined;
    current = current[part];
  }
  return current;
}

File: projhub/lib/compiler/template.py
from __future__ import annotations
import re


def resolve_template(template: str, config: dict) -> str:
    def replace(match: re.Match) -> str:
        key = match.group(1).strip()
        val = _get_nested(config, key)
        return str(val) if val is not None else match.group(0)

    return re.sub(r"\{\{([^}]+)\}\}", replace, template)


def _get_nested(obj: dict, key: str):
    parts = key.split(".")
    current = obj
    for part in parts:
        if not isinstance(current, dict):
            return None
        current = current.get(part)
    return current
